extensions/machine-learning-services/src/common/processService.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as childProcess from 'child_process';
+
+export class ProcessService {
+
+	public async execScripts(exeFilePath: string, scripts: string[], outputChannel?: vscode.OutputChannel): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
```

Comment:
Any particular reason you're wrapping this in a promise? It's already an async function so there shouldn't be any problems moving this code to be directly called by the method.

Reply:
Oh you need the callbacks for the exit handler. Hmm, you may want to add some sort of timeout here in case the child process gets stuck and never exits. 

---
In reply to: [358517448](https://github.com/microsoft/azuredatastudio/pull/8622#discussion_r358517448) [](ancestors = 358517448)

Reply:
which one is a asyn function? I think I'm missing something 

---
In reply to: [358517448](https://github.com/microsoft/azuredatastudio/pull/8622#discussion_r358517448) [](ancestors = 358517448)